### PR TITLE
K-fold cross validation

### PR DESCRIPTION
**Pull request by GeneZC**
k-fold cross validation as requested by #48 and #32

### CODE REVIEW

File: cross_val.py [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+            embedding_matrix = build_embedding_matrix(
+                word2idx=tokenizer.word2idx,
+                embed_dim=opt.embed_dim,
+                dat_fname='{0}_{1}_embedding_matrix.dat'.format(str(opt.embed_dim), opt.dataset))
+            self.model = opt.model_class(embedding_matrix, opt).to(opt.device)
+
+        dataset = ABSADataset(opt.dataset_file['train'], tokenizer)
+        datalen = len(dataset)
+        dataset = [dataset[i*datalen//opt.fold:(i+1)*datalen//opt.fold] for i in range(opt.fold)]
+        self.dataset = [(list(itertools.chain.from_iterable(dataset[:i]+dataset[i+1:])), dataset[i]) for i in range(opt.fold)]
```
Comment (bbcfox) [inline cross_val.py]: Add shuffle?

Comment (GeneZC) [reply cross_val.py]: I've added shuffle option.

**Comment by songyouwei** [comment]
This cross-validation approach may not be the most appropriate one. 
I think the implementation of K-fold validation may refer to [this approach](https://github.com/howardhsu/BERT-for-RRC-ABSA/blob/4477a904ee1f821d0d3c0515254c859c6854ee3c/src/run_asc.py#L125).

**Comment by GeneZC** [comment]
@songyouwei  You're right, I may fix it later.

**Pull request merged**